syntax = "proto3";

package realestate;

service RealEstateService {
    
    rpc StreamPropertyListings(Empty) returns (stream Property);
    rpc StreamPropertyUpdates(Empty) returns (stream PropertyUpdate);
}

// Property Listing Structure
message Property {
    int32 id = 1;
    string brokered_by = 2;
    string status = 3;
    double price = 4;
    int32 bed = 5;
    int32 bath = 6;
    double acre_lot = 7;
    string street = 8;
    string city = 9;
    string state = 10;
    string zip_code = 11;
    double house_size = 12;
    string prev_sold_date = 13;
}

message PropertyUpdate {
    int32 id = 1;
    double price = 2;
    string status = 3;
}

// Empty message for the request
message Empty {}